kartothek/io/dask/_update.py

```diff
@@ -35,6 +37,40 @@ def _hash_bucket(df: pd.DataFrame, subset: List[str], num_buckets: int):
     return df.assign(**{_KTK_HASH_BUCKET: buckets.astype(f"uint{bit_width}")})
 
 
+def _pack_payload(partition: pd.DataFrame, group_key: List[str]) -> pd.DataFrame:
+    # TODO: benchmark how good plain parquet would do
+    try:
+        # Technically distributed is an optional dependency
+        from distributed.protocol import serialize_bytes
+    except ImportError:
+        _logger.warning(
+            "Shuffle payload columns cannot be compressed since distributed is not installed."
+        )
+        return partition
+
+    # This is not _optimal_ since dask internally will perform another groupby in shuffle-group
+    # Once / if this is merged upstream, we can get rid of this
+    res = partition.groupby(group_key, sort=False, observed=True).apply(serialize_bytes)
+    res.name = _PAYLOAD_COL
+    return res.reset_index()
+
+
+def _unpack_payload(partition: pd.DataFrame) -> pd.DataFrame:
+    try:
+        # Technically distributed is an optional dependency
+        from distributed.protocol import deserialize_bytes
+    except ImportError:
+        _logger.warning(
+            "Shuffle payload columns cannot be compressed since distributed is not installed."
+        )
+        return partition
+
+    return pd.concat(
+        partition[_PAYLOAD_COL].map(lambda part: deserialize_bytes(part)).values,
+        copy=False,
```

Comment:
does this work w/o `ignore_index=False` aka are we sure that the index is unique here?

Reply:
I thought about it and am not entirely sure how dask depends on these indices. I'll try to come up with a few test cases to check this.

Reply:
At least dask itself is sensitive to this (e.g. for `set_index`). I guess within the kartothek context we get away with `ignore_index=True` (We don't properly preserve DF indices everywhere, anyhow)